modin/pandas/base.py

```diff
@@ -3199,6 +3200,88 @@ def __getitem__(self, key):
         else:
             return self._getitem(key)
 
+    @_inherit_docstrings(pandas.DataFrame.xs, apilink="pandas.DataFrame.xs")
+    def xs(
+        self,
+        key,
+        axis=0,
+        level=None,
+        drop_level: bool = True,
+    ):
+        axis = self._get_axis_number(axis)
+        labels = self.axes[axis]
+
+        if isinstance(key, list):
+            # deprecated in pandas, to be removed in 2.0
+            warnings.warn(
+                "Passing lists as key for xs is deprecated and will be removed in a "
+                + "future version. Pass key as a tuple instead.",
+                FutureWarning,
+            )
+
+        if level is not None:
+            if not isinstance(labels, pandas.MultiIndex):
+                raise TypeError("Index must be a MultiIndex")
+            loc, new_ax = labels.get_loc_level(key, level=level, drop_level=drop_level)
+
+            # create the tuple of the indexer
+            _indexer = [slice(None)] * self.ndim
+            _indexer[axis] = loc
+            indexer = tuple(_indexer)
+
+            result = self.iloc[indexer]
+            setattr(result, self._pandas_class._get_axis_name(axis), new_ax)
+            return result
+
+        if axis == 1:
+            if drop_level:
+                return self[key]
+            index = self.columns
+        else:
+            index = self.index
+
+        if isinstance(index, pandas.MultiIndex):
+            loc, new_index = index._get_loc_level(key, level=0)
```

Comment:
Why not this function? This place breaks in tests, with some old pandas version (compatibility tests in Modin).
```suggestion
            loc, new_index = index.get_loc_level(key, level=0)
```

Reply:
Could you use `get_loc_level` here? It seems that `_get_loc_level` is missed in old pandas: https://github.com/modin-project/modin/actions/runs/3439448795/jobs/5736821658
```python
AttributeError: 'MultiIndex' object has no attribute '_get_loc_level'
```

Reply:
> Could you use get_loc_level here?

not if we want it to be correct.

Reply:
ideally we'd drop the ancient pandas.  alternatively we could a) check for _get_loc_level and fall back to get_loc_level and just live with this incorrectness or b) fall back to the current default_to_pandas implementation

Reply:
@mvashishtha do we need to continue to maintain compatibility with pandas for python 3.6 or was 0.17 the last release for this?

Reply:
We don't want to support python 3.6 any more, and I started to take out our support for it, but I don't know when I'll finish. It takes some effort just to remove it. Meanwhile for a quick fix we could stop testing for it in CI and remove the documentation saying we support it-- is that okay?

Reply:
@mvashishtha yes, it's ok for me. Thanks! Can you show where we need to change the documentation about `xs` function for compatibility mode?

@jbrockmendel just skip the failed test in compatibility mode as in https://github.com/modin-project/modin/blob/daf68bb97b9ee58099137477afdf06ad1d4c75a4/modin/pandas/test/test_io.py#L820

Reply:
We can leave a note under "known issues" here: https://github.com/modin-project/modin/blob/41e32e9edcd1e6755d6f959e573f807a75c7ae7e/docs/supported_apis/older_pandas_compat.rst#L21

Reply:
if 3.6 is getting dropped (and im assuming thats the blocker for dropping old pandas), then why not just wait to merge this until after that is done.  no need to introduce additional kludges just to remove them in a few weeks

Reply:
@jbrockmendel I'm ok to wait

Reply:
@jbrockmendel 3.6 is dropped, need rebase